protocol-tests/src/main/java/fixtures/header/implementation/HeadersImpl.java

```diff
@@ -278,2183 +243,1300 @@
-     * @return a DynamicRequest where customizations can be made before sent to the service.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Void> paramDatetimeRfc1123Async(String scenario, RequestOptions requestOptions, Context context) {
-        return paramDatetimeRfc1123WithResponseAsync(scenario, requestOptions, context)
+    public Mono<Void> paramDatetimeRfc1123Async(RequestOptions requestOptions, Context context) {
+        return paramDatetimeRfc1123WithResponseAsync(requestOptions, context)
                 .flatMap((Response<Void> res) -> Mono.empty());
     }
 
     /**
      * Send a post request with header values "scenario": "valid", "value": "Wed, 01 Jan 2010 12:34:56 GMT" or
      * "scenario": "min", "value": "Mon, 01 Jan 0001 00:00:00 GMT".
-     *
-     * @param scenario Send a post request with header values "scenario": "valid" or "min".
-     * @return a DynamicRequest where customizations can be made before sent to the service.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public void paramDatetimeRfc1123(String scenario, RequestOptions requestOptions) {
-        paramDatetimeRfc1123Async(scenario, requestOptions).block();
+    public void paramDatetimeRfc1123(RequestOptions requestOptions) {
+        paramDatetimeRfc1123Async(requestOptions).block();
     }
 
     /**
      * Send a post request with header values "scenario": "valid", "value": "Wed, 01 Jan 2010 12:34:56 GMT" or
      * "scenario": "min", "value": "Mon, 01 Jan 0001 00:00:00 GMT".
-     *
-     * @param scenario Send a post request with header values "scenario": "valid" or "min".
-     * @return a DynamicRequest where customizations can be made before sent to the service.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<Void> paramDatetimeRfc1123WithResponse(
-            String scenario, RequestOptions requestOptions, Context context) {
-        return paramDatetimeRfc1123WithResponseAsync(scenario, requestOptions, context).block();
+    public Response<Void> paramDatetimeRfc1123WithResponse(RequestOptions requestOptions, Context context) {
+        return paramDatetimeRfc1123WithResponseAsync(requestOptions, context).block();
     }
 
-    /**
-     * Get a response with header values "Wed, 01 Jan 2010 12:34:56 GMT" or "Mon, 01 Jan 0001 00:00:00 GMT".
-     *
-     * @param scenario Send a post request with header values "scenario": "valid" or "min".
-     * @return a DynamicRequest where customizations can be made before sent to the service.
-     */
+    /** Get a response with header values "Wed, 01 Jan 2010 12:34:56 GMT" or "Mon, 01 Jan 0001 00:00:00 GMT". */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<Void>> responseDatetimeRfc1123WithResponseAsync(
-            String scenario, RequestOptions requestOptions) {
+    public Mono<Response<Void>> responseDatetimeRfc1123WithResponseAsync(RequestOptions requestOptions) {
```

Comment:
Seems like the `Header Parameters` javadoc is still missing

Reply:
Fixed